lib/commands/install.js

```diff
@@ -13,7 +13,15 @@ module.exports = Command.extend({
     { name: 'save', type: Boolean, default: false, aliases: ['S'] },
     { name: 'save-dev', type: Boolean, default: true, aliases: ['D'] },
     { name: 'save-exact', type: Boolean, default: false, aliases: ['E', 'exact'] },
-    { name: 'yarn', type: Boolean, description: 'Use --yarn to enforce yarn usage, or --no-yarn to enforce npm' }, // no default means use yarn if the blueprint has a yarn.lock
+    { name: 'yarn', type: Boolean, description: 'This is deprecated. Use --package-manager instead.' }, // no default means use yarn if the blueprint has a yarn.lock
+    {
+      name: 'package-manager',
+      type: String,
+      description:
+        'Use this option to force the usage of a specific package manager. ' +
+        'By default, ember-cli will try to detect the right package manager ' +
+        'from any lockfiles that exist in your project. Supported values: npm, yarn, pnpm',
+    },
```

Comment:
Hmm, seems like we don't have to deprecate `--yarn`. I was thinking we could do something like:

https://github.com/ember-cli/ember-cli/blob/70ae16c5eb972076580b23e9c22579234749862e/lib/commands/build.js#L16

```suggestion
    {
      name: 'package-manager',
      type: String,
      description:
        'Use this option to force the usage of a specific package manager. ' +
        'By default, ember-cli will try to detect the right package manager ' +
        'from any lockfiles that exist in your project. Supported values: npm, yarn, pnpm',
      aliases: [{ yarn: 'yarn' }, { pnpm: 'pnpm' }],
    },
```